lib/app/features/feed/create_story/providers/story_viewing_provider.dart

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'dart:math';
+
+import 'package:ion/app/features/feed/create_story/data/models/story.dart';
+import 'package:ion/app/features/feed/create_story/data/models/story_viewer_state.dart';
+import 'package:ion/app/features/feed/views/pages/feed_page/components/stories/mock.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'story_viewing_provider.g.dart';
+
+@riverpod
+class StoryViewingController extends _$StoryViewingController {
+  @override
+  StoryViewerState build() => const StoryViewerState.initial();
+
+  Future<void> loadStories() async {
+    state = const StoryViewerState.loading();
+
+    try {
+      final stories = await _fetchStories();
+
+      if (stories.isEmpty) {
+        state = const StoryViewerState.error(message: 'No stories available');
+        return;
+      }
+
+      state = StoryViewerState.ready(
+        stories: stories,
+        currentIndex: 0,
+      );
+    } catch (e) {
+      state = StoryViewerState.error(message: e.toString());
+    }
+  }
+
+  void moveToNextStory() {
+    state.mapOrNull(
```

Comment:
You use `mapOrNull` a lot in this class. Does the `whenOrNull` semantically fit better? Because `map` is usually used to convert (map) some value to another value. But here you just do some logic without returning a value.

Reply:
Indeed, thanks. Updated